src/test/java/com/company/linkedlist/MiddleNodeTest.java

```diff
@@ -0,0 +1,39 @@
+package com.company.linkedlist;
+
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.stream.Stream;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class MiddleNodeTest {
+    @ParameterizedTest
+    @MethodSource("testDataProvider")
+    void testReverseLinkedList(ListNode head, int expectedResult) {
```

Comment:
Test Method name says for different problem

Reply:
changed